gserver/src/steps.handler.ts

```diff
@@ -127,7 +127,8 @@ export default class StepsHandler {
         //Step text could be placed between '/' symbols (ex. in JS) or between quotes, like in Java
         const stepStart = stepRegExSymbol ? `(${stepRegExSymbol})` : `(\/|'|"|\`)`;
         //Our step could contain any symbols, except of our 'stepStart'. Use \3 to be sure in this
-        const stepBody = stepRegExSymbol ? `([^${stepRegExSymbol}]+)` : '([^\\3]+)';
+        //ref to RegEx Example: https://regex101.com/r/mS1zJ8/1
+        const stepBody = stepRegExSymbol ? `([^${stepRegExSymbol}]+)` : `((?:(?=(?:\\\\)*)\\\\.|.)*?)`;
```

Comment:
Did you test this case with some `stepRegExSymbol`  options set? (like when `stepRegExSymbol = '` and step contains some apostrophe.

Reply:
No, I did not, I presumed if that was set, my changes would have been ignored. 

> Are proposing instead of these changes, that setting `stepRegExSymbol = '` would solve our case? Sorry a bit confused what you're asking for here. :)

Reply:
By default, `/ \` " '` symbols are used as step regex start and end. Sometimes users would like to exclude some of these symbols or use some other ones. `stepRegExSymbol` is an option, that allows ensuring this (ex. `stepRegExSymbol='` means, that all the other symbols - `/ \` "` - would be excluded).

So, I meant, that your updates are fixing step getting in case of `stepRegExSymbol` option is not set, but not fixing step getting in case,  when `stepRegExSymbol` option is set (ex. step=`Given('I\'ll be back)` and stepRegExSymbol=`'`). Looks like it would be better to fix this and, maybe, add some tests.

Reply:
I see, yes, I had ignored that setting since I was so focused on the original `[^\\3]+`.

So just to give you a run down of the reasonal of the changes I made. My original intent was just to switchthe regex to use `*?` instead of `+` in which case I should have adjusted it for both cases (which I missed). [See this commit](https://github.com/alexkrechik/VSCucumberAutoComplete/pull/434/commits/68b194fa85a2e9257e22fdf87ed50df712987320).

But when I ran the unit tests against that, I discovered it couldn't address the escape characters, which the `+` did inadvertently since it took the last `'` to resolve the test case.

So I switched the RegEx to be specific to escape any specific `\'` characters, but still be lazy on the fetch which `*?` gives us. But by switching to that RegEx, now we aren't specific to the third parameter (the start/close symbol) `\\3` or whatever the specific `stepRegExSymbol is. So in my latest commit I just removed that logic. 

I also added more testing to ensure we're capturing these issues and it seems it's all working now (hopefully). 😄